index.js

```diff
@@ -1,11 +1,55 @@
 /* eslint-disable no-empty-function */
 const { Plugin } = require('powercord/entities');
-const { typing } = require('powercord/webpack');
-
 module.exports = class SilentType extends Plugin {
     startPlugin() {
-        this.oldStartTyping = typing.startTyping;
-        typing.startTyping = () => {};
+        powercord.api.commands.registerCommand({
+            command: 'silent-type',
+            description: 'Keep your typing to yourself!',
+            usage: '{c} [--on|--off|--status]',
+            executor: (args) => ({
+                send: false,
+                result:  this.handler(args)
+            })
+        })
+        // Magic
+        this.callRefresh()
+    }
+     handler(args) {
+        args = args.map(a => a.toLowerCase())
+        if (args.includes('--off') && args.includes('--on') || args.includes('--on') && args.includes('status')) {
+            return "Oops! Instead of receiving one flag I got both!\nSend only one flag next time"
+        } else if (args.includes('--on')) {
+            if (this.settings.get('on', true)) {
+                return "I am already turned on\nIf you would like to turn me off use the `--off` flag"
+            }
+            this.settings.set('on')
+            this.callRefresh()
+            return "Turned On"
+        } else if (args.includes('--off')) {
+            if (!this.settings.get('on')) {
+                return "I am already turned off\nIf you would like to turn me on use the `--on` flag"
+            }
+            this.settings.set('on')
```

Comment:
`this.settings.set(false)`

For feasibility sake I'd like to change I/O to a Boolean. Easier to manipulate if a settings panel is added later too